uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/viewmodel/CardDetailsCheckoutViewModel.kt

```diff
@@ -140,7 +140,7 @@ internal class CardDetailsCheckoutViewModel(
                 ),
                 actionButtonState = currentState.actionButtonState.updateIsEnabled(newValue = false),
             )
-        } else if (!cardCheckoutScreenValidator.isCardNumberValid(cardNumberValue)) {
+        } else if (!cardCheckoutScreenValidator.isCardNumberValid(cardNumberValue) || !cardCheckoutScreenValidator.isCardSchemaSupported(cardNumberValue, currentState.allowedCardSchemes)) {
```

Comment:
just a suggestion:  can we make the validator validate the card by only one method and inside that one, we do both the validations instead of exposing 2 methods here which will affect the unit tests for the view Model?